calculator_test.go

```diff
@@ -3,50 +3,43 @@ package calculator_test
 import (
 	"calculator"
 	"math/rand"
+	"strings"
 	"testing"
 	"time"
 )
 
-// GenerateNums it is a helper function
-// that produces a slice of two float numbers.
+// testCese represents input data and expected output
+// from the function under test.
+type testCase struct {
+	name        string
+	a           float64
+	b           float64
+	c           []float64
+	want        float64
+	expectedErr bool
+}
+
+// GenerateNums it is a helper function that produces a slice
+// of minimum two and maximum 100 float numbers.
 func GenerateNums() []float64 {
 	// Make generator to produce different
 	// numbers each time it runs (make it nondeterministic).
 	s := rand.NewSource(time.Now().UnixNano())
```

Comment:
In fact, we only need to seed the random generator once, at program startup. While the `init()` function is one way to do this, many people discourage the use of `init()` functions because they represent 'magical' behaviour which isn't necessarily obvious to the reader. An alternative way to do this which avoids `init()` is to create a new, seeded `rand.Rand` at package level, as follows:

```go
var random = rand.New(rand.NewSource(time.Now().UnixNano()))
```

We can then refer to `random.Intn()`, for example, rather than `rand.Intn()`.